pkg/plugins/external/init.go

```diff
@@ -40,16 +44,29 @@ func (p *initSubcommand) BindFlags(fs *pflag.FlagSet) {
 }
 
 func (p *initSubcommand) Scaffold(fs machinery.Filesystem) error {
+	configBytes, err := yaml.Marshal(p.config)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Scaffolding with config:", string(configBytes))
+
 	req := external.PluginRequest{
 		APIVersion: defaultAPIVersion,
 		Command:    "init",
 		Args:       p.Args,
+		Config:     string(configBytes),
 	}
 
-	err := handlePluginResponse(fs, req, p.Path)
+	err = handlePluginResponse(fs, req, p.Path)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (p *initSubcommand) InjectConfig(c config.Config) error {
+	p.config = c
+	return nil
+}
```

Comment:
However, to ensure that works could we add some tests to this one?

In the e2e test that we have now, could we not validate the config is filled and with the data that is expected?
Also, could we try to pass a value unto the config and see if that will be persisted in the PROJECT file?

See here how we update the config with the data to have it in tracked in the PROJECT file: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/deploy-image/v1alpha1/api.go#L221-L247

Note that this code is reponsable for we track: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/PROJECT#L8-L25